### ISSUE TITLE
BUZZER no longer works on Anet A6

### ISSUE THREAD

**Issue by JohnnyTheOne** [MarlinFirmware/Marlin#9804]
BUZZER no longer works on Anet A6. are there others who also have it?
I use Bugfix 1.1.x and 2.0.x

**Comment by thinkyhead** [comment]
When did it begin? It works ok in 1.1.8? New configs? `SPEAKER` enabled?

**Comment by JohnnyTheOne** [comment]
I do not know when it started and yes speaker enabled. wanted to test the buzzer
I have tested version 1.1.0 does not work either. it must be in pins_ANET_10.h or my board is broken

**Comment by thinkyhead** [comment]
Got an oscilloscope?

**Comment by JohnnyTheOne** [comment]
no sorry

**Comment by thinkyhead** [comment]
Terribly sorry. We know the buzzer code works. You have a hardware problem.

**Comment by JohnnyTheOne** [comment]
no hardware problem. I have tested pin 4 and it works
```
#define BEEPER_PIN 4 // 17
```
M300 S4000 P100

**Comment by thinkyhead** [comment]
Ah, so you have solved a bug in our pins definition for A6…?

**Comment by thinkyhead** [comment]
Can you verify that your display is either `REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER` or `ANET_FULL_GRAPHICS_LCD`? We incorporated the A8 configuration in advance of the A6, and received only one pins definition pertaining to both. Are you able to read the silkscreen on your board and see a version number?

**Comment by JohnnyTheOne** [comment]
no only that pin 17 does not work

**Comment by thinkyhead** [comment]
It's encouraging to know that Minimalism is alive and well. We'll follow up on this information and see what we can learn from other Anet A6 owners. Thanks much!

**Comment by JohnnyTheOne** [comment]
my display is ANET_FULL_GRAPHICS_LCD and I have also tested REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER in config
and I have the Anet V1.0 from 2016

**Comment by thinkyhead** [comment]
Bloody Pin 4! I hope this isn't one of those things I've heard before, meant to fix, and totally forgot.

**Comment by JohnnyTheOne** [comment]
pin 4 is also not a good option it disturbs the FAN PIN
pin 12 seems to work?

**Comment by thinkyhead** [comment]
Isn't your buzzer hard-wired to the board or the LCD? How are you swapping pins around?

**Comment by JohnnyTheOne** [comment]
buzzer is hard wired to the LCD. more gambling that science and I know is wrong

**Comment by benlye** [comment]
According to the board schematics the buzzer on the A6 LCD is connected to pin 10 on the LCD connector.  Pin 10 on the LCD connector is in turn connected to digital pin 17 on the Atmega1284p.  I've confirmed this with a meter.

https://raw.githubusercontent.com/ralf-e/ANET-3D-Board-V1.0/master/ANET3D_Board_Schematic.png
https://raw.githubusercontent.com/ralf-e/ANET-3D-Board-V1.0/master/ANET3D_LCD12864_Board_Schematic.png

**Comment by benlye** [comment]
I just confirmed that the buzzer on my Anet A6 LCD works correctly with the current pin definition of 
`#define BEEPER_PIN       17` in bugfix-2.0.x.

I guess your board or LCD has a hardware issue.

**Comment by thinkyhead** [comment]
Thanks for the input, Ben. Now I know who to ask about A6 issues should there ever be any in the future. This is probably the last one we'll ever get, though.

**Comment by keithluneau** [comment]
I just wanted to comment on this since it was helpful in diagnosing an issue with my printer. I have an A8 (same board as the A6), and I've installed the "Anet full graphics display" from the A6 onto it. I'm running Marlin 1.1.8. The beeper/buzzer on the display did not work. I was unable to generate any sounds using an M300 command, and had no feedback when selecting menu items. I landed here after searching a bit, and was happy to find the schematics posted for both the main board and LCD board. I worked through them and verified that they are correct (at least relevant to the buzzer) for the v1.5 Anet mainboard that I have. Looking at the code and the datasheet for the Atmega1284p, everything seems to be in order. I first verified that the Atmega was generating a signal on pin 20 (internal pin 17 as defined in the code), and then that the BC817 on the LCD board was closing and connecting the buzzer to ground as expected. 5V was present on the buzzer's positive pin. The next step was to remove the buzzer from the LCD board and replace it with a known working one. When doing so I found that the solder pad was physically not connected to the LCD board and that the trace leading to it was broken. It appears to have been overheated when assembled, or just poor quality PCB to begin with. Probably both... lol Once removed I found that the buzzer worked fine with 5V applied, so I scraped away the coating on the trace next to the buzzer's pin and soldered it back in making sure I had a good connection, and it works fine now. So just to verify, the pin for the buzzer is properly defined in the code, and the Anet full graphics display uses a piezoelectric buzzer, not a speaker.
